[UPDATE BONUS GUIDE] Important update of the Bisq bonus guide (#1075)
raspibolt/raspibolt: guide/bonus/bitcoin/bisq.md

---

## Requirements

* Bitcoin Core

---

## Introduction

The guide will show you how to:

> 1. Configure BTC Core allowing Bisq to run its SPV wallet and whitelisting the P2P connection

Reviewer: what is an SPV wallet?

1. Install Bisq on your personal computer
1. Connect Bisq to your Bitcoin Core own node in your local network or via remote Tor and depending on your OS personal computer
1. Securely set up Bisq

---

## Installation

* Download Bisq on your personal computer using the appropriate binary for your OS: [https://bisq.network/downloads/](https://bisq.network/downloads/){:target="_blank"}

* Verify the downloaded binary following [Bisq guidelines](https://bisq.wiki/Downloading_and_installing){:target="_blank"}

* Once you've verified the integrity of the downloaded Bisq binary, install it on your personal computer but do NOT launch Bisq yet!

---

### Configure Firewall

> * Configure the firewall to allow incoming requests to BTC Core from the local network.

Author: Fixed!

  ```sh

>   $ sudo ufw allow 8333 comment 'allow BTC Core' 

Author: Fixed!

>   $ sudo ufw allow 8333 comment 'allow Bitcoin Core' 

Author: Fixed!

  ```

### Configure Bitcoin Core

> For Bisq to connect to your Bitcoin Core node, the bloom filters have to be activated and whitelisted the P2P connection.

Author: Fixed!

* Open `bitcoin.conf` to add `peerbloomfilters=1` under `"# Bitcoin daemon"` to activate bloom filters

  ```sh
  $ sudo nano /data/bitcoin/bitcoin.conf
  ```

  ```sh
  # Filters
  peerbloomfilters=1   # for Bisq
  ```

* Add the following line under `"whitelist=download@127.0.0.1       # for Electrs"` line to whitelist our own P2P connection.

  ```sh

>   whitelist=bloomfilter@0.0.0.0   # for Bisq

Reviewer 2: Bloom filters has known DoS vectors, especially against nodes with not so powerful CPU, with this we will be ok to serve them to any other node on a Bitcoin P2P network. Limiting this to LAN would be simple (`bloomfilter@192.168.0.0/16`), but then remote connecting via Tor below will not work. Don't have solution for that (SSH tunneling with port forwarding to home network from remote laptop could be a potential solution), but maybe that should be noted here. Especially for people that maybe need only LAN connections, not remote. https://bitcoinops.org/en/topics/transaction-bloom-filtering/

Author: should we then forget about the local connection for Bisq which needs bloom filters enabled? :(

Reviewer 2: Local (LAN) is not a problem at all (limiting to 192.168.0.0/16 handles that), potential problem is remote to bitcoind .onion address.

Reviewer 3: @kristapsk @twofaktor I tested with the bloomfilter LAN restriction and Bisq can still remote connect to Bitcoin Core via Tor.

I have:
* `bitcoind.conf`
```
peerbloomfilters=1 # for Bisq
whitelist=bloomfilter@192.168.0.0/16
```

* Bisq "Use Tor for Bitcoin network" ticked
* Bisq using my Bitcoin Core Tor onion addess & port 8333
* I can see the "Bitcoin network peers: 1" at the bottom and the block height in Bisq matches the ongoing block height.

So looks like we could restrict the bloomfilter to LAN and still be able to access both locally (via clearnet) and remotely (via Tor).

Reviewer 3: @kristapsk do you think going with these settings is OK for the time being? (bloomfilter restricted to LAN, still able to connect via Tor)

If so, I think this PR is ready to be merged.

>   whitelist=bloomfilter@0.0.0.0   # for Bisq

Reviewer 3: ```suggestion
  bloomfilter@192.168.0.0/16   # for Bisq
```
As discussed above

Author: did you want to say `whitelist="bloomfilter@192.168.0.0/16`??

Reviewer 3: yes, I updated the code suggestion. It was a typo.

Author: Fixed!

Reviewer 3: could you also test if you can do a remote connection via Tor with this whitelist option restricted to LAN only?

Author: yeah sure, It's working for me too, look at this:

![bisq_onion](https://user-images.githubusercontent.com/89636253/201618828-46519ae3-fa09-41fa-a84a-e1c66d2b2551.png)

  ```

> * By default `bind` configuration of BTC Core is `"0.0.0.0"`. If you decide to connect Bisq from another personal computer in your local network, comment or delete `"bincoind=127.0.0.1"`. Save and exit.

Author: Fixed!

> * By default `bind` configuration of BTC Core is `"0.0.0.0"`. If you decide to connect Bisq from another personal computer in your local network, comment or delete `"bincoind=127.0.0.1"`. Save and exit.

Author: I was following the [official development docs] (https://bisq.wiki/Connecting_to_your_own_Bitcoin_node) section "Connecting to another Bitcoin node", maybe it's better to put `bind=192.168.0.0/16` instead of `bind=0.0 .0.0 `, but I didn't try it this way. Anyway, if you don't touch port 8333 on your NAT router to make it remotely accessible, I don't think there is a security risk. In the Raspibolt guide, we never touch router configuration. 
Anyway, could you try to change `bind=0.0.0.0` to `bind=192.168.0.0/16` to see if the connection to Bisq is still working in the same way?

On your personal computer where you installed Bisq, depending on your OS

**For Linux:**

* Open a command line terminal, we will start Bisq with two flags that will force it to connect to our own node only. Bisq should connect to your node on startup.

From local network connection, replace `192.168.X.X:8333` with your own node IP address.
  
  ```sh
  $ /opt/bisq/bin/Bisq -btcNodes=192.168.X.X:8333 -useTorForBtc=false
  ```

From remote connection, replace `123...abc.onion:8333` with your own Bitcoin Core .onion address that you obtained above.
  
  ```sh
  $ /opt/bisq/bin/Bisq -btcNodes=123...abc.onion:8333 -useTorForBtc=true
  ```

* Wait a few minutes until Bisq is up to date with the current state of the blockchain and go back to "Settings" > "Network info" to check that only your own node local IP address or onion address is listed in the first table.

* Check that the "Bitcoin network peers" counter at the bottom right of the window is equal to 1.

**For MacOS:**

* Open a command line terminal, we will start Bisq with two flags that will force it to connect to our own node only.

> From local network connection, replace `192.168.X.X:8333` with your own node IP address.

Reviewer 3: ```suggestion
* From local network connection, replace `192.168.X.X:8333` with your own node IP address.
```

a nit

Author: Fixed!

  
  ```sh
  $ Bisq -btcNodes=192.168.X.X:8333 -useTorForBtc=false
  ```

> From remote connection, replace `123...abc.onion:8333` with your own Bitcoin Core .onion address that you obtained above.

Reviewer 3: ```suggestion
* From remote connection, replace `123...abc.onion:8333` with your own Bitcoin Core .onion address that you obtained above.
```

Author: Fixed!

  $ sudo systemctl restart bitcoind
  ```

### Uninstall FW configuration

> * Delete firewall rule with the comment 'allow Fulcrum SSL' identifying the number of the rule

Reviewer 2: ```suggestion
* Delete firewall rule with the comment 'allow Bitcoin Core' identifying the number of the rule
```

Author: Fixed!

  ```sh
  $ sudo ufw status numbered
  ```

  ```sh
  Status: active
     To                         Action      From
     --                         ------      ----

>   [X] 8333                       ALLOW IN    Anywhere            # allow BTC Core

Reviewer 2: ```suggestion
  [X] 8333                       ALLOW IN    Anywhere            # allow Bitcoin Core
```

Author: Fixed!
